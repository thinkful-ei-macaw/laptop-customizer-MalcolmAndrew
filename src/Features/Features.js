import React from 'react';
import './Features.css';
import Options from '../Options/Options'

class Features extends React.Component {
  render(){
    const features = Object.keys(this.props.list).map((feature, idx) => {
      
      const featureHash = feature + '-' + idx;
      
      return (
        <Options 
          key={featureHash}
          feature={this.props.list[feature]} 
          featureName={feature}
          onChange={this.props.onChange} 
          hash={featureHash} 
          selected={this.props.selected[feature]}/>
      );
    });

    return (
      <form className="main__form">
        <h2>Customize your laptop</h2>
        {features}
      </form>
    )
  }
}

export default Features