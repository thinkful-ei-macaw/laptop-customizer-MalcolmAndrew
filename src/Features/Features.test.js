import React from 'react';
import ReactDOM from 'react-dom'
import Features from './Features';
import renderer from 'react-test-renderer'

const SELECTED = {
    Processor: {
      name: '17th Generation Intel Core HB (7 Core with donut spare)',
      cost: 700
    },
    'Operating System': {
      name: 'Ubuntu Linux 16.04',
      cost: 200
    },
    'Video Card': {
      name: 'Toyota Corolla 1.5v',
      cost: 1150.98
    },
    Display: {
      name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
      cost: 1500
    }
}

const FEATURES = {
  Processor: [
    {
      name: '17th Generation Intel Core HB (7 Core with donut spare)',
      cost: 700
    },
    {
      name: 'Professor X AMD Fire Breather with sidewinder technology',
      cost: 1200
    }
  ],
  "Operating System": [
    {
      name: 'Ubuntu Linux 16.04',
      cost: 200
    },
    {
      name: 'Bodhi Linux',
      cost: 300
    }
  ],
  "Video Card": [
    {
      name: 'Toyota Corolla 1.5v',
      cost: 1150.98
    },
    {
      name: 'Mind mild breeze 2000',
      cost: 1345
    }
  ],
  Display: [
    {
      name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
      cost: 1500
    },
    {
      name: '15.3" HGTV (3840 x 2160) Home makeover edition',
      cost: 1400
    },
  ]
};



describe('<Features />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Features list={FEATURES} selected={SELECTED}/>, div)
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders this UI as expected', () => {
    const tree = renderer.create(<Features list={FEATURES} selected={SELECTED}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  
})


