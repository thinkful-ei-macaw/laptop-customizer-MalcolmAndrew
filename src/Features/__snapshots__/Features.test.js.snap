// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Features /> renders this UI as expected 1`] = `
<form
  className="main__form"
>
  <h2>
    Customize your laptop
  </h2>
  <fieldset
    className="feature"
  >
    <legend
      className="feature__name"
    >
      <h3>
        Processor
      </h3>
    </legend>
    <div
      className="feature__item"
    >
      <input
        checked={true}
        className="feature__option"
        id="\\"name\\":\\"17th-Generation-Intel-Core-HB-(7-Core-with-donut-spare)\\"\\"cost\\":700"
        name="Processor"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"17th-Generation-Intel-Core-HB-(7-Core-with-donut-spare)\\"\\"cost\\":700"
      >
        17th Generation Intel Core HB (7 Core with donut spare)
         (
        $700.00
        )
      </label>
    </div>
    <div
      className="feature__item"
    >
      <input
        checked={false}
        className="feature__option"
        id="\\"name\\":\\"Professor-X-AMD-Fire-Breather-with-sidewinder-technology\\"\\"cost\\":1200"
        name="Processor"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"Professor-X-AMD-Fire-Breather-with-sidewinder-technology\\"\\"cost\\":1200"
      >
        Professor X AMD Fire Breather with sidewinder technology
         (
        $1,200.00
        )
      </label>
    </div>
  </fieldset>
  <fieldset
    className="feature"
  >
    <legend
      className="feature__name"
    >
      <h3>
        Operating System
      </h3>
    </legend>
    <div
      className="feature__item"
    >
      <input
        checked={true}
        className="feature__option"
        id="\\"name\\":\\"Ubuntu-Linux-16.04\\"\\"cost\\":200"
        name="Operating-System"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"Ubuntu-Linux-16.04\\"\\"cost\\":200"
      >
        Ubuntu Linux 16.04
         (
        $200.00
        )
      </label>
    </div>
    <div
      className="feature__item"
    >
      <input
        checked={false}
        className="feature__option"
        id="\\"name\\":\\"Bodhi-Linux\\"\\"cost\\":300"
        name="Operating-System"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"Bodhi-Linux\\"\\"cost\\":300"
      >
        Bodhi Linux
         (
        $300.00
        )
      </label>
    </div>
  </fieldset>
  <fieldset
    className="feature"
  >
    <legend
      className="feature__name"
    >
      <h3>
        Video Card
      </h3>
    </legend>
    <div
      className="feature__item"
    >
      <input
        checked={true}
        className="feature__option"
        id="\\"name\\":\\"Toyota-Corolla-1.5v\\"\\"cost\\":1150.98"
        name="Video-Card"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"Toyota-Corolla-1.5v\\"\\"cost\\":1150.98"
      >
        Toyota Corolla 1.5v
         (
        $1,150.98
        )
      </label>
    </div>
    <div
      className="feature__item"
    >
      <input
        checked={false}
        className="feature__option"
        id="\\"name\\":\\"Mind-mild-breeze-2000\\"\\"cost\\":1345"
        name="Video-Card"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"Mind-mild-breeze-2000\\"\\"cost\\":1345"
      >
        Mind mild breeze 2000
         (
        $1,345.00
        )
      </label>
    </div>
  </fieldset>
  <fieldset
    className="feature"
  >
    <legend
      className="feature__name"
    >
      <h3>
        Display
      </h3>
    </legend>
    <div
      className="feature__item"
    >
      <input
        checked={true}
        className="feature__option"
        id="\\"name\\":\\"15.6\\"-UHD-(3840-x-2160)-60Hz-Bright-Lights-and-Knobs\\"\\"cost\\":1500"
        name="Display"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"15.6\\"-UHD-(3840-x-2160)-60Hz-Bright-Lights-and-Knobs\\"\\"cost\\":1500"
      >
        15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs
         (
        $1,500.00
        )
      </label>
    </div>
    <div
      className="feature__item"
    >
      <input
        checked={false}
        className="feature__option"
        id="\\"name\\":\\"15.3\\"-HGTV-(3840-x-2160)-Home-makeover-edition\\"\\"cost\\":1400"
        name="Display"
        onChange={[Function]}
        type="radio"
      />
      <label
        className="feature__label"
        htmlFor="\\"name\\":\\"15.3\\"-HGTV-(3840-x-2160)-Home-makeover-edition\\"\\"cost\\":1400"
      >
        15.3" HGTV (3840 x 2160) Home makeover edition
         (
        $1,400.00
        )
      </label>
    </div>
  </fieldset>
</form>
`;
